src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -772,4 +784,183 @@ public void delete(Long programmingExerciseId, boolean deleteBaseReposBuildPlans
     public List<ProgrammingExercise> findAllWithBuildAndTestAfterDueDateInFuture() {
         return programmingExerciseRepository.findAllByBuildAndTestStudentSubmissionsAfterDueDateAfterDate(ZonedDateTime.now());
     }
+
+    /**
+     * Search for all programming exercises fitting a {@link PageableSearchDTO search query}. The result is paged,
+     * meaning that there is only a predefined portion of the result returned to the user, so that the server doesn't
+     * have to send hundreds/thousands of exercises if there are that many in Artemis.
+     *
+     * @param search The search query defining the search term and the size of the returned page
+     * @return A wrapper object containing a list of all found exercises and the total number of pages
+     */
+    public SearchResultPageDTO<ProgrammingExercise> getAllOnPageWithSize(final PageableSearchDTO<String> search) {
+        var sorting = Sort.by(ProgrammingExercise.ProgrammingExerciseSearchColumn.valueOf(search.getSortedColumn()).getMappedColumnName());
```

Comment:
I agree to Thilo that we need to take the permissions into account here. This is quite important so that instructors can only import programming exercises they have access to.

Please use the Principal to retrieve the current user with his groups and check if `exercise.course.instructorGroupName` is included. Ideally this is already performed in the database call. If this is not so easy doable, you can also do this in Java and simply remove programming exercises from the page which do not fulfill the criteria

Reply:
I am now checking whether the user's groups contain the group of an exercise and filter the result in the DB call accordingly